### ISSUE TITLE
This plugin breaks when using typescript-eslint 7.8.0.

### ISSUE THREAD

**Issue by cpojer** [sweepline/eslint-plugin-unused-imports#77]
See https://github.com/typescript-eslint/typescript-eslint/pull/8640 for more information. Just opening an issue here for cross-tracking and to figure out a good path forward so that `eslint-plugin-unused-imports` can be made to work again.

**Comment by JoshuaKGoldberg** [comment]
FWIW I'm somewhat sure you can get the equivalent variable AST node using a combination of:

1. The built-in ESLint `sourceCode.getNodeByRangeIndex` method to jump to the AST from a range index: https://eslint.org/docs/latest/extend/custom-rules#accessing-the-source-code -> `getNodeByRangeIndex`
2. Climbing up `node.parent` until you find the kind of node you want (`VariableDeclarator`?)

**Comment by cpojer** [comment]
@JoshuaKGoldberg: Option 2 is what this plugin already does, but `typescript-eslint` stopped providing `node` on the reported problem, which is what brings us here.

**Comment by cpojer** [comment]
Thanks so much for fixing this @sweepline!

**Comment by marcosdemelo** [comment]
In my project, I did have this error:

```shell
TypeError: Cannot destructure property 'parent' of 'node' as it is undefined.
Occurred while linting /home/user/Documents/repositories/project/services/utils.spec.js:1
Rule: "unused-imports/no-unused-imports"
    at exports.unusedImportsPredicate (/home/rafael/Documents/repositories/facil123/node_modules/eslint-plugin-unused-imports/lib/rules/predicates.js:28:13)
    at Object.value [as report] (/home/rafael/Documents/repositories/facil123/node_modules/eslint-rule-composer/lib/rule-composer.js:144:25)
    at Program:exit (/home/rafael/Documents/repositories/facil123/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-unused-vars.js:301:33)
    at ruleErrorHandler (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.leaveNode (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/node-event-generator.js:349:14)
    at CodePathAnalyzer.leaveNode (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:824:23)
    at /home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/linter.js:1113:32
    at Array.forEach (<anonymous>)
    at runRules (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/linter.js:1106:15)
    at Linter._verifyWithoutProcessors (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/linter.js:1355:31)
    at Linter._verifyWithoutProcessors (/home/rafael/Documents/repositories/facil123/node_modules/eslint-plugin-eslint-comments/lib/utils/patch.js:166:36)
    at Linter._verifyWithConfigArray (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/linter.js:1807:21)
    at Linter.verify (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/linter.js:1437:65)
    at Linter.verifyAndFix (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/linter/linter.js:2068:29)
    at verifyText (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/cli-engine/cli-engine.js:254:48)
    at CLIEngine.executeOnFiles (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/cli-engine/cli-engine.js:834:28)
    at ESLint.lintFiles (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/eslint/eslint.js:551:23)
    at Object.execute (/home/rafael/Documents/repositories/facil123/node_modules/eslint/lib/cli.js:421:36)
    at async main (/home/rafael/Documents/repositories/facil123/node_modules/eslint/bin/eslint.js:152:22)
```

Thanks by PR!

**Comment by marcosdemelo** [comment]
In my project, I did have this error:

```shell
TypeError: Cannot destructure property 'parent' of 'node' as it is undefined.
Occurred while linting /home/user/Documents/repositories/project/services/utils.spec.js:1
Rule: "unused-imports/no-unused-imports"
    at exports.unusedImportsPredicate (/home/user/Documents/repositories/project/node_modules/eslint-plugin-unused-imports/lib/rules/predicates.js:28:13)
    at Object.value [as report] (/home/user/Documents/repositories/project/node_modules/eslint-rule-composer/lib/rule-composer.js:144:25)
    at Program:exit (/home/user/Documents/repositories/project/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-unused-vars.js:301:33)
    at ruleErrorHandler (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /home/user/Documents/repositories/project/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.leaveNode (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/node-event-generator.js:349:14)
    at CodePathAnalyzer.leaveNode (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:824:23)
    at /home/user/Documents/repositories/project/node_modules/eslint/lib/linter/linter.js:1113:32
    at Array.forEach (<anonymous>)
    at runRules (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/linter.js:1106:15)
    at Linter._verifyWithoutProcessors (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/linter.js:1355:31)
    at Linter._verifyWithoutProcessors (/home/user/Documents/repositories/project/node_modules/eslint-plugin-eslint-comments/lib/utils/patch.js:166:36)
    at Linter._verifyWithConfigArray (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/linter.js:1807:21)
    at Linter.verify (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/linter.js:1437:65)
    at Linter.verifyAndFix (/home/user/Documents/repositories/project/node_modules/eslint/lib/linter/linter.js:2068:29)
    at verifyText (/home/user/Documents/repositories/project/node_modules/eslint/lib/cli-engine/cli-engine.js:254:48)
    at CLIEngine.executeOnFiles (/home/user/Documents/repositories/project/node_modules/eslint/lib/cli-engine/cli-engine.js:834:28)
    at ESLint.lintFiles (/home/user/Documents/repositories/project/node_modules/eslint/lib/eslint/eslint.js:551:23)
    at Object.execute (/home/user/Documents/repositories/project/node_modules/eslint/lib/cli.js:421:36)
    at async main (/home/user/Documents/repositories/project/node_modules/eslint/bin/eslint.js:152:22)
```

Thanks by PR!
